packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-variant-management.ts

```diff
@@ -0,0 +1,88 @@
+import type FlexCommand from 'sap/ui/rta/command/FlexCommand';
+
+import { QuickActionContext, SimpleQuickActionDefinition } from '../../../cpe/quick-actions/quick-action-definition';
+import { areManifestChangesSupported, prepareManifestChange } from './utils';
+import ListReportComponent from 'sap/suite/ui/generic/template/ListReport';
+
+import { SimpleQuickActionDefinitionBase } from '../simple-quick-action-base';
+import Component from 'sap/ui/core/Component';
+
+export const ENABLE_VARIANT_MANAGEMENT_IN_TABLES_CHARTS = 'enable-variant-management-in-tables-charts';
+
+const CONTROL_TYPES = ['sap.f.DynamicPage'];
+
+/**
+ * Quick Action for enabling table filtering using table personalization settings.
+ */
+export class EnableListReportVariantManagementQuickAction
+    extends SimpleQuickActionDefinitionBase
+    implements SimpleQuickActionDefinition
+{
+    private isPageSmartVariantManagementEnabled = false;
+    private ownerComponent: ListReportComponent;
+    readonly forceRefreshAfterExecution = true;
+
+    constructor(context: QuickActionContext) {
+        super(
+            ENABLE_VARIANT_MANAGEMENT_IN_TABLES_CHARTS,
+            CONTROL_TYPES,
+            'QUICK_ACTION_ENABLE_TABLES_AND_CHARTS_VARIANT_MANAGEMENT',
+            context,
+            [
+                {
+                    run: () => {
+                        if (this.ownerComponent) {
+                            this.isPageSmartVariantManagementEnabled = this.ownerComponent.getSmartVariantManagement();
```

Comment:
Enablement check should not modify state, it should only check if action should be enabled or not. This should be done in initialize method.

Also we need to read pending changes to get the correct current value.

Reply:
fixed